### ISSUE TITLE
Common handling of service schedule lifetime model

### ISSUE THREAD

**Issue by xi-yang** [esnet/sense-rm#14]
This is to normalize the modeling and handling of scheduled L2 connection as applicable to SENSE-RM. 

The feature requirements described in https://github.com/esnet/SENSE-Testbed/issues/92#issue-820370980 and https://github.com/esnet/SENSE-Testbed/issues/92#issuecomment-789205102.

**Comment by xi-yang** [comment]
This will be similar to Items 1, 2 and 4 for the SENSE-O NSIDriver feature https://github.com/esnet/StackV/issues/1150.

Run experiment with a NSIDriver backed domain, such as icair mren8700 on  [sense-o-dev.es.net](https://sense-o-dev.es.net:8443/StackV-web/portal/) to observe the modeling.

**Comment by R-Jimenez** [comment]
@jmacauley here are the example TTLs you wanted. I stood up a fresh stub ONSA server to keep things focused and captured three snapshots:
1. `base.txt` - Fresh domain without any connections.
2. `scheduled.txt` - An instance (`1c67088b-0883-421d-bf98-9fb02213a67a`) has been created and scheduled, and is not yet activated.
3. `active.txt` - Another instance (`c5a4398b-2ad0-4235-9474-09cab217d8b0`) has been created, scheduled, and is now active. At this time the previous instance has expired, and you can see the elements associated from it are no longer in the model.

[base.txt](https://github.com/esnet/sense-rm/files/12053236/base.txt)
[scheduled.txt](https://github.com/esnet/sense-rm/files/12053235/scheduled.txt)
[active.txt](https://github.com/esnet/sense-rm/files/12053234/active.txt)

Let me know if you need/want any more references, or have any questions on the model code.

**Comment by R-Jimenez** [comment]
In addition to the above examples, went ahead and did a side-by-side. After reviewing a SENSE-RM model (`urn:ogf:network:ultralight.org:2013`) with a scheduled profile, I believe the adjustments we discussed in our meeting should happen easily enough. A non-exhaustive diff list from my review:

* `NetworkStatus` resources are brand new. Specifics and case-definitions are contained within the [NSIService ref class](https://github.com/esnet/StackV/blob/Milestone-M12-production_operational/StackV-ejb/src/main/java/net/maxgigapop/mrs/driver/nsi/NSIService.java).
* `Lifetime` resources seem to already be handled well. With a scheduled service I saw correct lifetime specified. There is some minor string formatting differences but nothing functional. (`ns0:end "2023-07-19T18:31:25Z" ;` vs `ns2:end "2023-07-23T16:54:16.000+0000"^^xsd:string ;`)
* `SwitchingSubnet`, `BidirectionalPort` (for VLAN ports), and `BandwidthService` resources need to add a `hasNetworkStatus` relation.
* Current `existDuring` relations look fine.

**Comment by R-Jimenez** [comment]
Below are a couple code snippets to help guide the porting process:

https://github.com/esnet/StackV/blob/7dc371645bed7bd5f38a614748a74ddc0e885f18/StackV-ejb/src/main/java/net/maxgigapop/mrs/driver/nsi/NSIService.java#L1084-L1137

Here is the network status moderation function block. It returns a simple string enum (currently defined as `ACTIVATING("activating"), ACTIVATED("activated"), ACTIVATE_ERROR("activate-error"),
    DEACTIVATING("deactivating"), DEACTIVATED("deactivated"), DEACTIVATE_ERROR("deactivate-error"),
    UNKNOWN("unknown"), ERROR("error");`).

The method takes in a `ConnectionStatesType` object, a class that is generated via the same WSDL that the sense-rm uses. This means that you should be able to copy this function over without any issue. In our code (which generates models on-demand) it is used like this:

```
              String networkStatus = parseNetworkStatus(conn.getConnectionStates());
              Resource resNetworkStatus = RdfOwl.createResource(model, metadata.getUri() + ":status",
                      Mrs.NetworkStatus);
              resNetworkStatus.addProperty(Mrs.type, "dataplane");
              resNetworkStatus.addProperty(Mrs.value, networkStatus);
              resSwitchingSubnet.addProperty(Mrs.hasNetworkStatus, resNetworkStatus);
```

From my review of the sense-rm code I see that you have a service (or multiple?) dealing with querying NSAs for reservations, so within a loop over said reservations you would end up just replacing `conn` above with `reservation`, and attaching the returned `networkStatus` to the summary object I assume you are passing to the model builder.

To go further than this I'd appreciate a pointer to where exactly you are constructing the model elements from said query summaries, or let me know if you have any further questions.

**Comment by jmacauley** [comment]
Cleaning up unclosed issues.  This issue was closed with commit cd80b107c71d4578d06c23aba6ede5a5ca820487.
